File: c_sharp/Professor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace c_sharp
{
    class Professor : Person, UniversityMember
    {
        private int id, publications;
        private static int professorID;

        public Professor()
        {
            id = ++professorID;
        }

        public void getData()
        {
            Console.WriteLine(id + " Name: " + name + " Age: " + age + " Publications: " + publications);
        }

        public void setData()
        {
            Console.WriteLine("New Professor");
            Console.Write("Set name: ");
            name = Console.ReadLine();

            Console.WriteLine("\n");

            Console.Write("Set age: ");
            age = Convert.ToInt32(Console.ReadLine());

            Console.WriteLine("\n");

            Console.Write("Set publications: ");
            publications = Convert.ToInt32(Console.ReadLine());
        }
    }
}

File: java/src/main/java/org/example/Professor.java
package org.example;

import java.util.Scanner;

public class Professor extends Person implements UniversityMember {
    private int _id = 0, publications = 0;
    private static int professorID = 0;
    Scanner scanner = new Scanner(System.in);

    Professor() {
        _id = ++professorID;
    }

    @Override
    public void setData() {
        System.out.println("new Professor");
        try {
            System.out.print("Insert age: ");
            age = Integer.parseInt(scanner.nextLine());
        }catch (NumberFormatException e){
            e.printStackTrace();
        }

        System.out.print("Insert name: ");
        name = scanner.nextLine();

        try {
            System.out.print("Insert publications: ");
            publications = Integer.parseInt(scanner.nextLine());
        }catch (NumberFormatException e){
        e.printStackTrace();
    }
    }

    @Override
    public void getData() {
        System.out.println(_id + " Name:" + name + " Age:" + age + " Publications: " + publications);
    }
}
